vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -119,6 +120,16 @@ func CloneTLSConfig(cfg *tls.Config) *tls.Config {
 	if cfg == nil {
 		return &tls.Config{}
 	}
+
+	if strings.HasPrefix(runtime.Version(), "go1.8") {
+		// make a copy of the config
+		tlsConfigCopy := cfg.Clone()
```

Comment:
yeah, this won't compile in go1.7... you need to do conditional compilation with two new files:

clone.go:
```go
// +build go1.8
func CloneTLSConfig(cfg *tls.Config) *tls.Config {
  cfgCopy := cfg.Clone()
  cfgCopy.Clone()
  return cfgCopy
}
```

clone_17.go:
```go
// +build !go1.8

func CloneTLSConfig(cfg *tls.Config) *tls.Config {
   ... existing impl ...
}
```